package/environments/base.js

```diff
@@ -22,7 +22,9 @@ const config = require('../config')
 
 const getLoaderList = () => {
   const result = new ConfigList()
-  Object.keys(rules).forEach(key => result.append(key, rules[key]))
+  Object.keys(rules)
+    .filter(rule => rule !== 'nodeModules' || (rule === 'nodeModules' && config.compile_dependencies))
+    .forEach(key => result.append(key, rules[key]))
```

Comment:
You would have to test for yourself, but this might be a bit easier to read:

```js
if (config.compile_dependencies == false) {
  delete rules.nodeModules;
}

result = { ...result, ...rules };
// const result = { ...new ConfigList(), ...rules }; // I don't think this would work
```

Reply:
Made it less terrible here: 26e0891

Reply:
I would still go with a simple `if`, there's no reason to iterate if we know exactly which value we don't want. The `keys`/`forEach`/`filter`/`append` stuff adds enough to the complexity without additional ternaries.

```js
if (config.compile_dependencies == false) {
  delete rules.nodeModules;
}
```